From fcbfa2ca616b46ff1d0ed5d2b89ee6f998a98070 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 7 Aug 2022 14:28:16 +0200
Subject: [PATCH 014/389] bluetooth: bcm: Restore drive_rts_on_open = true
 behavior on bcm20702a1

This is what it was on v5.5. It doesn't fix the UART issues though.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/bluetooth/hci_bcm.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/bluetooth/hci_bcm.c b/drivers/bluetooth/hci_bcm.c
index d7e0b75db8a6..02495c988214 100644
--- a/drivers/bluetooth/hci_bcm.c
+++ b/drivers/bluetooth/hci_bcm.c
@@ -1554,8 +1554,13 @@ static struct bcm_device_data cyw55572_device_data = {
 	.max_autobaud_speed = 921600,
 };
 
+static struct bcm_device_data bcm20702a1_device_data = {
+	.drive_rts_on_open = true,
+	.no_early_set_baudrate = true,
+};
+
 static const struct of_device_id bcm_bluetooth_of_match[] = {
-	{ .compatible = "brcm,bcm20702a1" },
+	{ .compatible = "brcm,bcm20702a1", .data = &bcm20702a1_device_data },
 	{ .compatible = "brcm,bcm4329-bt" },
 	{ .compatible = "brcm,bcm4330-bt" },
 	{ .compatible = "brcm,bcm4334-bt" },
-- 
2.35.3

